lib/ansible/plugins/action/copy.py

```diff
@@ -539,6 +539,9 @@ def run(self, tmp=None, task_vars=None):
             changed = changed or module_return.get('changed', False)
 
         if module_executed and len(source_files['files']) == 1:
+
+            module_return['invocation'].update({'content': content})
+
```

Comment:
Talked about this in slack.  I believe we  should use "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER" instead of content as we considered display of content a security problem waaay back in 1.2.0 (the first release after content was introduced) https://github.com/ansible/ansible/issues/2884  This was the fix proposed by mpdehaan.  I'm not sure that it worked then, but at some point (maybe 2.0 or 2.1) we started getting the displayed arguments from invocation which would have fixed that if it had been broken before.

So I think that the issue history shows that the module has a policy of consciously hiding the value of the content parameter for security reasons.  We *can* decide that we want to change that policy (content is really no different in this regard than lineinfile, for instance) but since it was introduced as a security fix we need to subject the change to a deprecation cycle, prominent release notes of the change,  and possibly a toggle.

Reply:
agreed, changed name of PR and i'll look into adding a deprecation notice